### PR TITLE
add replicaset+pod support, and fix table display issue

### PR DESCRIPTION
**Pull request by cdlliuy**
# Changes

- :gift:  add replicaset and pods support
- :bug: fix the display issue when the table content is very long

This is still a part of the initial code

### CODE REVIEW

File: pkg/command/diagnose/service.go [hunk 1]
```diff
@@ -160,14 +160,14 @@ func createKeyInfoTable(table Table, f *Fetcher) error {
...
-		return fmt.Errorf("Can't fetch any ImageCache CR for the target service")
+		return fmt.Errorf("Can't fetch any ImageCache CR for the target knative service")
 	}
 	imgCacheResource := NewPrintableResource(3, "image", f.Images.ObjectMeta.Name, "--",
 		WithVerboseType(verbose))
 	table.AddMuitpleRows(imgCacheResource.DumpResource())
 
 	if f.KPA == nil {
-		return fmt.Errorf("Can't fetch any ImageCach CR for the target service")
+		return fmt.Errorf("Can't fetch any ImageCach CR for the target knative service")
```
Comment (maximilien) [inline pkg/command/diagnose/service.go:170]: Typo: `ImageCache`

File: pkg/command/diagnose/service.go [hunk 2]
```diff
@@ -180,7 +180,7 @@ func createKeyInfoTable(table Table, f *Fetcher) error {
 	table.AddMuitpleRows(kpaResource.DumpResource())
 
 	if f.Metrics == nil {
-		return fmt.Errorf("Can't fetch any ImageCach CR for the target service")
+		return fmt.Errorf("Can't fetch any ImageCach CR for the target knative service")
```
Comment (maximilien) [inline pkg/command/diagnose/service.go:183]: Same as above

File: pkg/command/diagnose/service.go [hunk 3]
```diff
@@ -226,14 +226,14 @@ func createKeyInfoTable(table Table, f *Fetcher) error {
 	table.AddMuitpleRows(routesResource.DumpResource())
 
 	if f.RoutesSVC == nil {
```
Comment (maximilien) [inline pkg/command/diagnose/service.go:228]: Extra space between “service” and “CR”

File: pkg/command/diagnose/service.go [hunk 4]
```diff
@@ -75,7 +75,7 @@ knative-diagnose service <ksvc-name> -ns <namespace>\n`)
 func dumpToTables(f *Fetcher, verbose string) error {
 
 	if f.KSVC == nil {
-		return fmt.Errorf("Can't fetch any KSVC CR for the target service")
+		return fmt.Errorf("Can't fetch any KSVC CR for the target knative service")
```
Comment (maximilien) [inline pkg/command/diagnose/service.go:78]: I wonder if this GIANT `dumpToTables` function could be split into multiple smaller helper functions focused on one aspect of the table?

**Review by maximilien — DISMISSED** [review]
Left a few comments. Thanks for keeping this moving along... do we also have a e2e test(s) that runt this on latest code?

File: pkg/command/diagnose/service.go (hunk 4 above)
Comment (cdlliuy) [reply pkg/command/diagnose/service.go:78]: yes, Max, it should be.  
This PR is only to make the current `demo` code base works .

I am thinking of refactor it completely to setup a good basis for more people to join.

File: pkg/command/diagnose/service.go (hunk 1 above)
Comment (cdlliuy) [reply pkg/command/diagnose/service.go]: fixed

**Comment by cdlliuy** [comment]
@maximilien ,  I didn't add e2e test on this yet, since I am thinking of a refactor on current `demo` code level soon to make the kn-diag structure more flexible.  I will share for the refactor design soon.
> Left a few comments. Thanks for keeping this moving along... do we also have a e2e test(s) that runt this on latest code?

**Comment by rhuss** [comment]
@cdlliuy we aligned the plugin with the latest dependencies (thanks @markusthoemmes ), there is one minor conflict which needs to be resolved (the client added a `context.Context()` as first argument to every client API call).

Sorry for any inconvenience but resolving the conflict shouldn't be that hard.

**Comment by cdlliuy** [comment]
@maximilien  @rhuss , the conflict is resolved :-)

**Comment by rhuss** [comment]
Thanks ! @maximilien if you are fine with the PR, feel free to approve via the GitHub UI (this should trigger the merge)

**Review by rhuss — APPROVED** [review]
Approving.

@maximilien quick reminder, for `knative-sandbox` projects, `/approved` is not good enough, you need to "Approve" via the GitHub UI. 

We are probably changing this back soon, to align with the `knative` organisation again, and allow self-approval.

**Pull request merged**